### PR TITLE
Dockerfile - upgrade nccl version and install ucx to fix bug in cuda 12.4 docker file

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
fix bug in cuda 12.4 docker file

**Major Revision**
- upgrade nccl due to OOM bug in nccl v2.20 graph mode
- install ucx 1.16 for mutli thread support for mpi in ib-traffic

### CODE REVIEW

File: dockerfile/cuda12.4.dockerfile [hunk 1]
```diff
@@ -115,6 +115,23 @@ RUN cd /tmp && \
     mv amd-blis /opt/AMD && \
     rm -rf aocl-blis-linux-aocc-4.0.tar.gz
 
+# Install NCCL 2.23.4
+RUN cd /tmp && \
+    git clone -b v2.23.4-1 https://github.com/NVIDIA/nccl.git && \
+    cd nccl && \
+    make -j src.build && \
```
Comment (abuccts) [inline dockerfile/cuda12.4.dockerfile:122]: `make -j ${NUM_MAKE_JOBS} src.build` to avoid OOM?

File: dockerfile/cuda12.4.dockerfile [hunk 2]
```diff
@@ -115,6 +115,23 @@ RUN cd /tmp && \
...
+    make install && \
+    rm -rf /tmp/nccl
+
+# Install UCX v1.16.0 with multi-threading support
+RUN cd /tmp && \
+    wget https://github.com/openucx/ucx/releases/download/v1.16.0/ucx-1.16.0.tar.gz && \
+    tar xzf ucx-1.16.0.tar.gz && \
+    cd ucx-1.16.0 && \
+    ./contrib/configure-release-mt --prefix=/usr/local && \
+    make -j && \
```
Comment (abuccts) [inline dockerfile/cuda12.4.dockerfile:132]: same

**Review by yzygitzh — APPROVED** [review]

File: dockerfile/cuda12.4.dockerfile (hunk 1 above)
Comment (yukirora) [reply dockerfile/cuda12.4.dockerfile]: thanks, updated

File: dockerfile/cuda12.4.dockerfile (hunk 2 above)
Comment (yukirora) [reply dockerfile/cuda12.4.dockerfile]: thanks, updated

**Pull request merged**